gapic/src/main/com/google/gapid/widgets/TabComposite.java

```diff
@@ -149,39 +157,58 @@ protected boolean flushCache(Control control) {
           update();
           break;
         default:
-          updateHover(getElement().find(group, 0, e.x, e.y, false));
+          updateHover(find(e.x, e.y));
       }
     });
 
     addListener(SWT.MouseUp, e -> {
+      if (e.button == 1 && dragger == null) {
+        switch (mouseDown.type) {
+          case Tab:
+            if (mouseDown.folder.updateCurrent(mouseDown.tab.control)) {
+              listeners.fire().onTabShown(mouseDown.tab.info);
+            }
+            break;
+          default:
+            // Do nothing.
+        }
+      }
```

Comment:
Why was this moved out of the `MouseDown` handler? I think it makes more sense there, especially for the immediate mouse-down-drag on a non selected tab. I think it's fine to move it back, I don't see any code reason for it to have to be moved here.

Reply:
This was actually a fix for non-selected tabs. When MouseDown is received, it is not clear yet if the user wants to select or drag the tab, so the select is postponed to MouseUp if no drag happened. The current implementation selects the tab first before it is dragged, which doesn't allow to maintain the selected tab if the user wants to drag a non-selected tab. All other tab UIs that I tested keep the selected tab if a non-selected tab is dragged.

Reply:
I disagree. Both Chrome and VSCode activate tabs on mouse down not mouse up. I remember now that I did it this way on purpose. Please restore the original functionality.

Reply:
I didn't say that all other UIs activate tabs on mouse up, even though there are some, for example Eclipse. I said that in all other UIs after dragging a non-selected tab to a destination folder, the selected tab in the source folder will be the same as before. This is also what a user intuitively would expect and this is not happening in AGI.
As requested I have restored the original functionality and filed a bug. b/174851718